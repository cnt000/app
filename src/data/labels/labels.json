{
  "openMenuLabel": "apri il menu",
  "title": "Pungilandia",
  "titleDescription": "Piante grasse, strane ed insolite",
  "goToCartLabel": "vai al carrello",
  "copyright": "Copyright © 2020",
  "privacy": "privacy",
  "contact": "contatti",
  "company": "Azienda Agricola Pungilandia Rocchi Domenico",
  "companyAddress": "Via Isonzo 11, 65123 Pescara P.IVA 01203140684 CF RCCDNC59R30I712B\nIscrizione REA Pescara Nr.97422",
  "disclaimer": "Azienda in possesso Registro di Detenzione di Specie Vegetali Incluse\nNegli Allegati CITES A e B del Regolamento (CE) 338/97 del Consiglio del 9\nDicembre 1996 e Successive Modifiche e Attuazioni. Registro N.000001/2001\nVAB",
  "mailToLink": "mailto:info@pungilandia.com",
  "privacyLink": "http://www.pungilandia.com",
  "addToCart": "Aggiungi al carrello",
  "searchPlants": "Cerca piante",
  "searchPlantsImage": "http://www.pungilandia.com/php/images/lo/cotyl_mediop.jpg",
  "hintsFromPungilandiaTitle": "I Consigli di Pungilandia",
  "articleBy": "A cura del dott. Mimmo Rocchi",
  "articleContent": "consigliColtivazione.md",
  "mimmoImage": "http://www.pungilandia.com/media/MIMMO.JPG",
  "menuChiSiamo": "Chi siamo",
  "menuListaPiante": "Lista piante",
  "menuNovita": "Novità",
  "menuProdottiSpeciali": "Prodotti speciali",
  "menuConsigliPerColtivare": "Consigli per coltivare",
  "menuOrdini": "Ordini",
  "menuFotoShow": "Foto show",
  "menuEmail": "E-mail"
}
