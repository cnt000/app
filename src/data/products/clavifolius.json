{
  "productName": "Adromischus clavifolius",
  "productImage": "http://www.pungilandia.com/php/images/hi/adromischus-clavifolius.jpg",
  "vaseDiameter": "Vaso 24cm",
  "price": "Prezzo: â‚¬ 9,00"
}
