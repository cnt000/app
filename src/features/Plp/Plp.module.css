.plpContent {
  grid-area: content;
  padding: 0.4rem;
}

.products {
  display: flex;
  flex-wrap: wrap;
}

.product {
  height: auto;
  width: 25%;
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
}

.product .name {
  margin: 0.6rem auto;
  width: 300px;
  line-height: 1.2;
}

.addToCart {
  margin: 0 auto;
  margin: 1rem auto 2rem auto;
  width: 50%;
  height: 2rem;
}

.pageHeader {
  display: flex;
  justify-content: space-between;
  height: 3rem;
  margin: 1rem auto;
}

.pageHeader .title {
  font-family: 'Caveat Brush', sans-serif;
  font-size: 32px;
  color: var(--green);
}

.product .image {
  width: 50%;
}

.pageHeader .cercapiante span {
  margin-left: -2.5rem;
  position: relative;
  text-shadow: 1px 1px 2px var(--white), 0 0 0.2em var(--grey);
}

@media only screen and (min-width: 1025px) {
  .plp-content {
    grid-area: content;
    padding: 2rem;
  }

  .pageHeader .left {
    text-align: left;
  }

  .pageHeader {
    display: flex;
    justify-content: space-between;
    height: 5rem;
    margin: 0.6rem auto;
  }
}
