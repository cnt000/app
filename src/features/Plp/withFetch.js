import React from 'react';
import { useParams } from 'react-router-dom';
import useFetch from 'use-http';
import { conf } from '../../conf/url';
import { useQuery } from '../../hook/useQuery';

const withFetch = (Component) => ({ children, ...props }) => {
  const { id } = useParams();
  const page = id ? Math.max(id - 1, 0) : 0;
  const query = useQuery();
  const searchQuery = query.get('q');
  const urlProducts = `${conf.apiEndpoint}${conf.plpUrl}${page}/24`;
  const urlSearchProducts = `${conf.apiEndpoint}${conf.plpUrl.slice(
    0,
    -1,
  )}/${page}/24?q=${searchQuery}`;
  const url = searchQuery ? urlSearchProducts : urlProducts;

  const { loading, data = [] } = useFetch(url, { cacheLife: 1 }, [url]);

  return (
    <>
      {loading && 'Loading...'}
      {!loading && (
        <Component
          products={data}
          page={page}
          searchQuery={searchQuery}
          {...props}
        >
          {children}
        </Component>
      )}
    </>
  );
};

export default withFetch;
