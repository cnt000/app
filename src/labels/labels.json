{
  "openMenuLabel": "apri il menu",
  "title": "Pungilandia",
  "titleDescription": "Piante grasse, strane ed insolite",
  "goToCartLabel": "vai al carrello",
  "copyright": "Copyright Â© 2020",
  "privacy": "privacy",
  "contact": "contatti",
  "company": "Azienda Agricola Pungilandia Rocchi Domenico",
  "companyAddress": "Via Isonzo 11, 65123 Pescara P.IVA 01203140684 CF RCCDNC59R30I712B<br>Iscrizione REA Pescara Nr.97422",
  "disclaimer": "Azienda in possesso Registro di Detenzione di Specie Vegetali Incluse<br>Negli Allegati CITES A e B del Regolamento (CE) 338/97 del Consiglio del 9<br>Dicembre 1996 e Successive Modifiche e Attuazioni. Registro N.000001/2001<br>VAB"
}
