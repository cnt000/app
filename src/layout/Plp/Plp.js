import React, { useContext } from 'react';
import PropTypes from 'prop-types';

import LabelsContext from '../../features/labels';
import styles from './Plp.module.css';

const Product = ({ i, productName, productImage, price }) => {
  const { addToCart } = useContext(LabelsContext);
  return (
    <article key={`${productName}_${i}`} className={styles.product}>
      <a href="/item" className={styles.imageContainer}>
        <img className={styles.image} src={productImage} alt={productName} />
        <h1 className={styles.name}>{productName}</h1>
        <div>{price}</div>
      </a>
      <button className={styles.addToCart}>{addToCart}</button>
    </article>
  );
};

Product.propTypes = {
  i: PropTypes.number,
  productName: PropTypes.string,
  productImage: PropTypes.string,
  price: PropTypes.string,
  addToCart: PropTypes.string,
};

const Plp = ({ productName, productImage, price }) => {
  const { searchPlants, searchPlantsImage } = useContext(LabelsContext);
  return (
    <main className={styles.plpContent} role="main">
      <div className={styles.pageHeader}>
        <div className={styles.left}>
          <h2 className={styles.title}>Le nostre piante</h2>
          <span className={styles.lastmodified}>02/04/2020</span>
        </div>
        <div className={styles.cercapiante}>
          <a href="/search" title={searchPlants}>
            <img src={searchPlantsImage} alt={searchPlants} />
            <span>{searchPlants}</span>
          </a>
        </div>
      </div>
      <div className={styles.products}>
        {Array(12)
          .fill(' ')
          .map((_, i) => Product({ i, productName, productImage, price }))}
      </div>
    </main>
  );
};

Plp.propTypes = {
  productName: PropTypes.string,
  productImage: PropTypes.string,
  price: PropTypes.string,
  addToCart: PropTypes.string,
  searchPlants: PropTypes.string,
  searchPlantsImage: PropTypes.string,
};

export default Plp;
