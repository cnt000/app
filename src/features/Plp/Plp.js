import React, { Suspense } from 'react';
import PropTypes from 'prop-types';

import useIntersect from '../../hook/useIntersect';
import Product from '../Product/Product';
import EmptyResults from '../EmptyResults/EmptyResults';
import CercaPiante from '../CercaPiante/CercaPiante';
import MyErrorBoundary from '../ErrorBoundary/ErrorBoundary';
import withFetch from './withFetch';
import Pagination from '../Pagination/Pagination';
import styles from './Plp.module.css';

const Plp = ({ products }) => {
  const [ref, entry] = useIntersect({
    threshold: 0.1,
  });

  const hasResults =
    !products.error && !products.code && products.message !== 'not found';

  const MemoizedPagination = React.memo(
    () =>
      entry.intersectionRatio > 0.1 && (
        <MyErrorBoundary>
          <Suspense
            fallback={<div className={styles.paginationBox}>Loading...</div>}
          >
            <Pagination />
          </Suspense>
        </MyErrorBoundary>
      ),
  );

  return (
    <main className={styles.plpContent} role="main">
      <div className={styles.pageHeader}>
        <div className={styles.left}>
          <h2 className={styles.title}>Le nostre piante</h2>
        </div>
        <CercaPiante />
      </div>
      <div className={styles.products}>
        {!hasResults && <EmptyResults />}
        {hasResults && products.map((product, i) => Product({ ...product, i }))}
      </div>
      <div
        ref={ref}
        ratio={entry.intersectionRatio}
        className={styles.paginationBox}
      >
        <MemoizedPagination />
      </div>
    </main>
  );
};

Plp.propTypes = {
  products: PropTypes.array,
  Link: PropTypes.object,
};

export default withFetch(Plp);
