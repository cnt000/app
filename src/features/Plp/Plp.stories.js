import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import Plp from './Plp';
import labels from '../../data/labels/labels.json';
import products from '../../data/products/searchpage.json';
import LabelsContext from '../Labels';

export default {
  component: Plp,
  title: 'Plp',
};

export const Default = () => (
  <LabelsContext.Provider value={labels}>
    <Router>
      <Plp products={products} />
    </Router>
  </LabelsContext.Provider>
);
