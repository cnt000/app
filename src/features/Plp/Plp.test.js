import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Link } from 'react-router-dom';

import Plp from './Plp';
import LabelsContext from '../../features/Labels';
import labels from '../../data/labels/labels.json';
import products from '../../data/products/searchpage.json';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <LabelsContext.Provider value={labels}>
        <Router>
          <Plp products={products} />
        </Router>
      </LabelsContext.Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
