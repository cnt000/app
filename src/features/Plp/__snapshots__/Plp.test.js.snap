// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  className="plpContent"
  role="main"
>
  <div
    className="pageHeader"
  >
    <div
      className="left"
    >
      <h2
        className="title"
      >
        Le nostre piante
      </h2>
      <span
        className="lastmodified"
      >
        02/04/2020
      </span>
    </div>
    <div
      className="cercapiante"
    >
      <a
        href="/search"
        title="Cerca piante"
      >
        <img
          alt="Cerca piante"
          src="http://www.pungilandia.com/php/images/lo/cotyl_mediop.jpg"
        />
        <span>
          Cerca piante
        </span>
      </a>
    </div>
  </div>
  <div
    className="products"
  >
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/2078"
        onClick={[Function]}
      >
        <img
          alt="Adromischus  clavifolius    ⌀24"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/adromischus-clavifolius.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Adromischus  clavifolius    ⌀24
        </h1>
        <div>
          € 9,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/2049"
        onClick={[Function]}
      >
        <img
          alt="Adromischus  filicaulis     ⌀10"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/adromischu_filica.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Adromischus  filicaulis     ⌀10
        </h1>
        <div>
          € 9,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/2075"
        onClick={[Function]}
      >
        <img
          alt="Adromischus   marianae F. Herrei    ⌀10"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/adromischus-herrei.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Adromischus   marianae F. Herrei    ⌀10
        </h1>
        <div>
          € 38,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/2052"
        onClick={[Function]}
      >
        <img
          alt="Adromischus  sculdtianus juttae    ⌀10"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/adromischus schlud.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Adromischus  sculdtianus juttae    ⌀10
        </h1>
        <div>
          € 9,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/1750"
        onClick={[Function]}
      >
        <img
          alt="Adromischus  truncatus    5,⌀5"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/adrom_trun.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Adromischus  truncatus    5,⌀5
        </h1>
        <div>
          € 5,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/2053"
        onClick={[Function]}
      >
        <img
          alt="Adromischus  cristatus  zeyheri    ⌀10"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/adromi_cristzeye.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Adromischus  cristatus  zeyheri    ⌀10
        </h1>
        <div>
          € 7,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/2080"
        onClick={[Function]}
      >
        <img
          alt="Aeonium  arboreum atropurpureum    ⌀10"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/aeonium-arboreum.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Aeonium  arboreum atropurpureum    ⌀10
        </h1>
        <div>
          € 4,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/831"
        onClick={[Function]}
      >
        <img
          alt="Aeonium  Kiwi, ibrido di Aeonium haworthii    ⌀6"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/aeoni_tri.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Aeonium  Kiwi, ibrido di Aeonium haworthii    ⌀6
        </h1>
        <div>
          € 3,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/1694"
        onClick={[Function]}
      >
        <img
          alt="Agave   attenuata    ⌀8"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/agave_attenuata8.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Agave   attenuata    ⌀8
        </h1>
        <div>
          € 4,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/1273"
        onClick={[Function]}
      >
        <img
          alt="Agave  filifera    ⌀10"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/agave_filq.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Agave  filifera    ⌀10
        </h1>
        <div>
          € 3,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/2094"
        onClick={[Function]}
      >
        <img
          alt="Agave   lophantha tricolor    ⌀10"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/agave_lophanta.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Agave   lophantha tricolor    ⌀10
        </h1>
        <div>
          € 7,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/1378"
        onClick={[Function]}
      >
        <img
          alt="Agave  maculata    ⌀20"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/agave_maculata.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Agave  maculata    ⌀20
        </h1>
        <div>
          € 6,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/2090"
        onClick={[Function]}
      >
        <img
          alt="Agave  nigra    ⌀10"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/agave_nigra.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Agave  nigra    ⌀10
        </h1>
        <div>
          € 9,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/1403"
        onClick={[Function]}
      >
        <img
          alt="Agave  victoriae-reginae    ⌀17"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/agave victoria.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Agave  victoriae-reginae    ⌀17
        </h1>
        <div>
          € 18,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/1216"
        onClick={[Function]}
      >
        <img
          alt="Agave filifera    ⌀6"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/aga_fil5.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Agave filifera    ⌀6
        </h1>
        <div>
          € 3,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/1215"
        onClick={[Function]}
      >
        <img
          alt="Agave stricta    ⌀10"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/agave_stric.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Agave stricta    ⌀10
        </h1>
        <div>
          € 3,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/2111"
        onClick={[Function]}
      >
        <img
          alt="Albuca  humilis    5.⌀5"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/albuca_humilis.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Albuca  humilis    5.⌀5
        </h1>
        <div>
          € 4,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/2110"
        onClick={[Function]}
      >
        <img
          alt="Albuca  spiralis    5.⌀5"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/albuca5.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Albuca  spiralis    5.⌀5
        </h1>
        <div>
          € 4,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/1737"
        onClick={[Function]}
      >
        <img
          alt="Allaudia  procera  h ⌀30"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/allaudia_pro.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Allaudia  procera  h ⌀30
        </h1>
        <div>
          € 9,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/1790"
        onClick={[Function]}
      >
        <img
          alt="Aloe  aloe x gasteria    ⌀12"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/aloexgasteria.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Aloe  aloe x gasteria    ⌀12
        </h1>
        <div>
          € 6,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/1387"
        onClick={[Function]}
      >
        <img
          alt="Aloe  bakeri    ⌀6"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/aloe bakeri.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Aloe  bakeri    ⌀6
        </h1>
        <div>
          € 3,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/1436"
        onClick={[Function]}
      >
        <img
          alt="Aloe  descoingsii pepe    5.⌀5"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/aloe_pepe.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Aloe  descoingsii pepe    5.⌀5
        </h1>
        <div>
          € 3,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/1697"
        onClick={[Function]}
      >
        <img
          alt="Aloe  humilis     ⌀10"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/aloe_humilis.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Aloe  humilis     ⌀10
        </h1>
        <div>
          € 4,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/1281"
        onClick={[Function]}
      >
        <img
          alt="Aloe   marlotti    5,⌀5"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/aloe_marlo.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Aloe   marlotti    5,⌀5
        </h1>
        <div>
          € 2,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/2023"
        onClick={[Function]}
      >
        <img
          alt="Aloe  melanacantha    ⌀5"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/aloe_melana.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Aloe  melanacantha    ⌀5
        </h1>
        <div>
          € 4,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/1913"
        onClick={[Function]}
      >
        <img
          alt="Aloe  pink blush    ⌀6"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/aloe_pink.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Aloe  pink blush    ⌀6
        </h1>
        <div>
          € 2,50
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/2050"
        onClick={[Function]}
      >
        <img
          alt="Aloe  ramosissima    ⌀10"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/aloe_ramosisi.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Aloe  ramosissima    ⌀10
        </h1>
        <div>
          € 5,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/1495"
        onClick={[Function]}
      >
        <img
          alt="Aloe  rauhii    ⌀8"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/aloe_rauhii.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Aloe  rauhii    ⌀8
        </h1>
        <div>
          € 3,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/1310"
        onClick={[Function]}
      >
        <img
          alt="Aloe   squarrosa    ⌀8"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/aloe_sqarrosa.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Aloe   squarrosa    ⌀8
        </h1>
        <div>
          € 3,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/pdp/1575"
        onClick={[Function]}
      >
        <img
          alt="Aloe  variegata    ⌀10"
          className="image"
          loading="lazy"
          src="https://qp2dljrm.twic.pics/aloe_var10.jpg?twic=v1/cover=300x399.98999999999995/quality=100"
        />
        <h1
          className="name"
        >
          Aloe  variegata    ⌀10
        </h1>
        <div>
          € 4,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
  </div>
  <a
    href="/plp/NaN"
    onClick={[Function]}
  >
    Next page (
    NaN
    )
  </a>
</main>
`;
